pkg/rpc/writer.go

```diff
@@ -20,46 +21,56 @@ import (
 type OutputWriter struct {
 	sync.Mutex
 	*zap.SugaredLogger
-	*progressWriter
+	pw *progressWriter
+	bw *binaryWriter
 
 	out io.Writer
 }
 
-var _ io.Writer = (*OutputWriter)(nil)
-
 func NewOutputWriter(w http.ResponseWriter, r *http.Request) *OutputWriter {
 	w.Header().Set("Content-Type", "application/json")
 
 	httpWriter := ioutils.NewWriteFlusher(w)
 
 	// progressWriter will emit log output as progress messages.
 	progressWriter := &progressWriter{out: httpWriter}
+
+	// binaryWriter will emit binary chunks
+	binaryWriter := &binaryWriter{}
+
 	writeSyncer := zapcore.Lock(zapcore.AddSync(progressWriter))
 
 	// this logger has two sinks: stdout and the writeSyncer, wired to the HTTP
 	// response.
 	logger := logging.NewLogger(writeSyncer).With(zap.String("req_id", r.Header.Get("X-Request-ID")))
 
 	ow := &OutputWriter{
-		SugaredLogger:  logger.Sugar(),
-		out:            httpWriter,
-		progressWriter: progressWriter,
+		SugaredLogger: logger.Sugar(),
+		out:           httpWriter,
+		pw:            progressWriter,
+		bw:            binaryWriter,
```

Comment:
No need to store it, just create it and return it from `OutputWriter#BinaryWriter()`.

Reply:
Does it really matter? If we use the binary writer multiple times, we would get the same writer right now, and not create multiple values... I am not sure if it makes any difference.